tests/cpp/common/test_gpu_hist_util.cu

```diff
@@ -51,7 +51,7 @@ void TestDeviceSketch(const GPUSet& devices, bool use_external_memory = false) {
 
   // find the cuts on the GPU
   HistCutMatrix hmat_gpu;
-  (void)DeviceSketch(p, gpu_batch_nrows, (*dmat).get(), &hmat_gpu);
+  (void)DeviceSketch(p, gpu_batch_nrows, dmat->get(), &hmat_gpu);
```

Comment:
Could you also check that the correct row stride is returned?

Reply:
done